src/components/MessagesList/MessagesList.vue

```diff
@@ -451,14 +451,14 @@ export default {
 					// first time load, initialize important properties
 					this.$store.dispatch('setFirstKnownMessageId', {
 						token: this.token,
-						id: this.conversation.lastReadMessage,
+						id: this.conversation.lastMessage.id,
```

Comment:
this feels very wrong. the lastMessage does not have to do anything with the reading status?

So when there are "more" messages you don't start at the readmarker anymore?

Reply:
yes.

imagine a conversation that exists on the server:
- message A
- message B
- unread marker
- message C
- message D

Previously, when calling getOldMessages it would only retrieve message A and message B.
then lookForNewMessages would right away fetch message C and message D, as they already exist.

With the new behavior, getOldMessages will fetch all messages A to D, so lookForNewMessages can start polling right away. There's no point in having to wait for lookForNewMessages to fetch the "old" messages C and D.

Or did you have something else in mind ?
-

Reply:
imagine a conversation that exists on the server:

* message A
* message B
* unread marker
* message 1
* …
* message 300

getOldMessages only gets 200 messages. so you will get message 100-300 you are far of your readmarker and the first read message is not even loaded, so you don't know

Reply:
hmmm, should we then set firstKnown to be the unread marker but keep lastKnown to be the last message ? (or the other way around, not sure about the order)

Reply:
That will break everything.

the "known" are the loaded ones. if you set a gap in there it will not load those messages ever. If you make them overlap you will load the messages between twice.

Reply:
I've reverted this back and have pushed an alternative approach